### DISCUSSION TITLE
how to fine tuning model with with dataset (file json/csv..)

### DISCUSSION THREAD

**Discussion by luongnv275** [ggml-org/llama.cpp#419] [Q&A]

**Comment by jorditost** [comment]
+1

is it possible?

**Comment by strikoder** [comment]
I'm interested as well...

**Comment by ianscrivener** [comment]
IMO - yes it is possible with prompt tuning

The next questions (which I am not so clear on) are;
  (1) exactly how to achieve that (a specific use case and dataset would be required)
  (2) how good/accurate the results would be....
  
  Of course the tuning dataset would need to fit into context

**Comment by sb-child** [comment]
hi, i just found this [article](https://rentry.org/cpu-lora) about fine tuning model with llama.cpp

This is an example of dataset format in appendix B of that article:

```
<s>Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
{prompt}

### Response:
{response}
```
And add a `--sample-start "<s>"` to the training command

**Reply by redoC-A2k** [reply]
Hey there did you tried fine tuning ? I want to fine tune this model https://huggingface.co/TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF . 
with finetune utiltiy and wondering about how to make th training txt file .
If you could help that will be very helpful
